crates/stc_ts_file_analyzer/src/analyzer/types/mod.rs

```diff
@@ -1537,19 +1537,71 @@ impl Analyzer<'_, '_> {
     pub(crate) fn expand_intrinsic_types(&mut self, span: Span, ty: &Intrinsic) -> VResult<Type> {
         let arg = &ty.type_args;
 
-        match ty.kind {
-            IntrinsicKind::Uppercase | IntrinsicKind::Lowercase | IntrinsicKind::Capitalize | IntrinsicKind::Uncapitalize => {
-                match arg.params[0].normalize() {
-                    Type::Lit(LitType { lit: RTsLit::Str(s), .. }) => {
-                        let new_val = match ty.kind {
-                            IntrinsicKind::Uppercase => s.value.to_uppercase(),
-                            IntrinsicKind::Lowercase => s.value.to_lowercase(),
+        match self.normalize(None, Cow::Borrowed(&arg.params[0]), Default::default())?.as_ref() {
+            Type::Lit(LitType { lit: RTsLit::Str(s), .. }) => {
+                let new_val = match ty.kind {
+                    IntrinsicKind::Uppercase => s.value.to_uppercase(),
+                    IntrinsicKind::Lowercase => s.value.to_lowercase(),
+                    IntrinsicKind::Capitalize => {
+                        if s.value.is_empty() {
+                            "".into()
+                        } else {
+                            let mut res = String::new();
+                            let mut chars = s.value.chars();
+
+                            res.extend(chars.next().into_iter().flat_map(|v| v.to_uppercase()));
+                            res.push_str(chars.as_str());
+
+                            res
+                        }
+                    }
+                    IntrinsicKind::Uncapitalize => {
+                        if s.value.is_empty() {
+                            "".into()
+                        } else {
+                            let mut res = String::new();
+                            let mut chars = s.value.chars();
+
+                            res.extend(chars.next().into_iter().flat_map(|v| v.to_lowercase()));
+                            res.push_str(chars.as_str());
+
+                            res
+                        }
+                    }
```

Comment:
Can you turn this into a function?
I think it can be handled by changing `JsKeyword` to Atom.

Reply:
I think so. Do you want me to do it?

Reply:
I'll try. Can I put it on this PR?

Reply:
Yes, thank you!

Reply:
https://github.com/dudykr/stc/pull/295/commits/3b25c0af65f0e274264f3a4b3c0c6bb7c8cc4725
I made
Is `JsWord` <-> `Atom` conversion correct?